base/deprecated.jl

```diff
@@ -241,3 +241,13 @@ end
 cat_shape(dims, shape::Tuple{}) = () # make sure `cat_shape(dims, ())` do not recursively calls itself
 
 # END 1.6 deprecations
+
+# BEGIN 1.7 deprecations
+
+# the plan is to eventually overload getproperty to access entries of the dict
+@noinline function getproperty(x::Pairs, s::Symbol)
+    depwarn("use values(kwargs) and keys(kwargs) instead of kwargs.data and kwargs.itr", :getproperty, force=true)
```

Comment:
Why the `force=true`?

Reply:
We need a depwarn version system that allows better granularity, but generally force=false means a v2 (breaking) deprecation while force=true means a user bug. This PR adds a warning for a user bug.